test/gtest/wat-parser.cpp

```diff
@@ -0,0 +1,331 @@
+#include "gtest/gtest.h"
+
+#include "wat-parser-internal.h"
+
+using namespace wasm::WATParser;
+
+TEST(ParserTest, LexWhitespace) {
+  Token expected{"42"sv, IntTok{42, false}};
+
+  Lexer lexer(" 42\t42\n42\r42 \n\n\t 42 "sv);
```

Comment:
Different numbers would help debugging, I think, and also make the test more precise in theory.